backend/app/services/positions.py

```
@@ -509,9 +538,15 @@ def recompute_position_state(
     # Calendar fallback: any leg past expiration with no resolving trade is
     # treated as expired worthless. See module/function docstring (issue #134).
-    open_legs, calendar_close_at = _apply_calendar_close(
+    open_legs, calendar_close_at, calendar_consumed = _apply_calendar_close(
         open_legs, position.ticker, clock()
     )
+    for leg in calendar_consumed:
+        # Synthetic worthless-expiry: stamp the leg's expiration on the
+        # originating Trade row (issue #136). No real resolving trade exists
+        # to supply an ``opened_at``, so the expiration is the canonical
+        # close date — same value the position-level ``closed_at`` would use.
+        closes_to_write.append((leg.trade_id, leg.expiration))
```

Ruff (0.15.12)

[warning] Use `list.extend` to create a transformed list

(PERF401)